ush/ioda/bufr2ioda/bufr2ioda_gnssro_cosmic2.py

```diff
@@ -0,0 +1,804 @@
+            mask = np.isin(said, mission_said)
+
+            # MetaData
+            clonh_sat = clonh[mask]
+            clath_sat = clath[mask]
+            gclonh_sat = gclonh[mask]
+            gclath_sat = gclath[mask]
+            timestamp_sat = timestamp[mask]
+            stid_sat = stid[mask]
+            said_sat = said[mask]
+            siid_sat = siid[mask]
+            sclf_sat = sclf[mask]
+            ptid_sat = ptid[mask]
+            elrc_sat = elrc[mask]
+            seqnum2_sat = seqnum2[mask]
+            geodu_sat = geodu[mask]
+            heit_sat = heit[mask]
+            impp1_sat = impp1[mask]
+            imph1_sat = imph1[mask]
+            mefr1_sat = mefr1[mask]
+            pccf_sat = pccf[mask]
+            ref_pccf_sat = ref_pccf[mask]
+            bearaz_sat = bearaz[mask]
+            ogce_sat = ogce[mask]
+            qfro_sat = qfro[mask]
+            qfro2_sat = qfro2[mask]
+            satasc_sat = satasc[mask]
+            bnda1_sat = bnda1[mask]
+            arfr_sat = arfr[mask]
+            bndaoe1_sat = bndaoe1[mask]
+            arfroe_sat = arfroe[mask]
+            bndaot_sat = bndaot[mask]
+            arfrot_sat = arfrot[mask]
+
+            # Processing Center
+            ogce_sat = ogce[mask]
+
+            # QC Info
+            qfro_sat = qfro[mask]
+            qfro2_sat = qfro2[mask]
+            satasc_sat = satasc[mask]
+
+            # ObsValue
+            bnda1_sat = bnda1[mask]
+            arfr_sat = arfr[mask]
+
+            # ObsError
+            bndaoe1_sat = bndaoe1[mask]
+            arfroe_sat = arfroe[mask]
+
+            # ObsType
+            bndaot_sat = bndaot[mask]
+            arfrot_sat = arfrot[mask]
+
+            nobs = clath_sat.shape[0]
+            print(' ... Create ObsSpace for satid = ', sat)
+            print(' ... size location of sat mission = ', nobs)
+
+        # =====================================
+        # Create IODA ObsSpace
+        # Write IODA output
+        # =====================================
+
+        # Create the dimensions
+    if nobs > 0:
+        dims = {'Location': np.arange(0, nobs)}
+        print(' ... dim = ', nobs)
+    else:
+        dims = {'Location': nobs}
+        print(' ... dim = ', nobs)
+
+        # iodafile = f"{cycle_type}.t{hh}z.{data_type}.tm00.{data_format}.nc"
+        # iodafile = f"{cycle_type}.t{hh}z.{ioda_data_type}.{mission}.tm00.{data_format}.nc"
+    iodafile = f"{cycle_type}.t{hh}z.{ioda_data_type}_{mission}.tm00.nc"
+
+    OUTPUT_PATH = os.path.join(ioda_dir, iodafile)
+
+    print(' ... ... Create OUTPUT file:', OUTPUT_PATH)
+
+    path, fname = os.path.split(OUTPUT_PATH)
+    if path and not os.path.exists(path):
+        os.makedirs(path)
+
+    # Create IODA ObsSpace
+    obsspace = ioda_ospace.ObsSpace(OUTPUT_PATH, mode='w', dim_dict=dims)
+
+    # Create Global attributes
+    logger.debug(f" ... ... Create global attributes")
+    obsspace.write_attr('source_file', bufrfile)
```

Comment:
I think the global attributes need to be closer to the standards.
At a minimum, 
data_format -> dataOriginalFormatSpec
subsets -> platforms
data_type should be deleted altogether since it's redundant with "source"
data_provider -> dataProviderOrigin

I'm unsure of the others. but I would look at the global attributes tab of this document to help figure it out.
https://docs.google.com/spreadsheets/d/1fHb2QIHOwgp4d6plyn9Dj9HZFZsDRxcPaUV6hdTb5BU/edit?gid=46200857#gid=46200857

@emilyhcliu do you agree?

I won't mark each script.

Reply:
Since we are going to switch to the high-level approach, we can fix the global attribute in the high-level Python script later.